### PR TITLE
Chore: enable unit test coverage comment

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

I've switched our CI to a different Jest action that has a better coverage report and the same annotations as the previous one.

Also, I've deleted the pre-commit hook because it was really slow.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -18,14 +18,10 @@ jobs:
...
-      - name: Run tests
-        run: 'yarn test:ci'
-
-      - name: Comment in failing tests
-        uses: mattallty/jest-github-action@v1.0.3
-        if: failure()
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Annotations and coverage report
+        uses: ArtiomTr/jest-coverage-report-action@v2
```
Comment (usame-algan) [inline .github/workflows/test.yml:22]: Is there a reason we are not using Coveralls? It seems that we already have some projects on there: https://coveralls.io/github/safe-global

Comment (katspaugh) [reply .github/workflows/test.yml:22]: I don't see the point of depending on a 3rd-party service for something available locally.

File: package.json [hunk 2]
```diff
@@ -32,9 +32,6 @@
   "engines": {
     "node": ">=16"
   },
-  "pre-commit": [
```
Comment (schmanu) [inline package.json:35]: Should we discuss this? It feels out of scope of the coverage report.

**Review by schmanu — COMMENTED** [review]
One thing that this setup is missing is seeing the diff to the base branch.
Or will this happen once it's merged to `dev`?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -18,14 +18,10 @@ jobs:
...
         with:
-          test-command: 'echo unit tests already executed'
-          coverage-comment: false
+          skip-step: install
+          package-manager: yarn
+          test-script: yarn test:ci
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (katspaugh) [reply .github/workflows/test.yml:27]: It keeps a single comment already, those two other comments in this PR are due to a different configuration while I was testing it.

File: package.json (hunk 2 above)
Comment (katspaugh) [reply package.json:35]: Let's dicuss it, yes. I'll start a thread on Slack.

**Comment by katspaugh** [comment]
@schmanu this action runs Jest on both the target branch and the PR branch and is supposed to give you a diff. I'm not sure why it's not reflected in this PR. Maybe it needs to be merged to dev indeed.
